### PR TITLE
Security fixes

### PR DESCRIPTION
**Pull request by carakas**
## Type

- Security

## Pull request description

Fixes a couple of security issues with uploads (images, themes), and block a whole lot of extra extensions that we don't want in case they get through

Also makes sure the session id changes when logging in or out

Add checks for csrf tokens in backend widgets

### CODE REVIEW

File: .htaccess [hunk 1]
```diff
@@ -49,7 +49,27 @@ deny from env=stayout
 
     # forbidden folders
     RewriteRule .*\.php - [F]
-    RewriteRule .*\.twig - [F]
+    RewriteRule \.php5 - [F]
```
Comment (jessedobbelaere) [inline .htaccess:52]: Q: Any reason we use `.*\.php` and the other rules don't have the wildcard?

File: src/Backend/Modules/Extensions/Actions/UploadTheme.php [hunk 2]
```diff
@@ -242,6 +242,7 @@ private function getValidatedFilesList(ZipArchive $zip): array
             // We skip all the files that are outside of the theme folder or on the ignore list.
             if ($this->checkIfPathContainsIgnoredWord($fileName) ||
                 (!empty($this->parentFolderName) && mb_stripos($fileName, $this->parentFolderName) !== 0)
+                || mb_strpos(basename($fileName), '.') === 0 // ignore hidden files
```
Comment (jessedobbelaere) [inline src/Backend/Modules/Extensions/Actions/UploadTheme.php:245]:
```suggestion
            if ($this->checkIfPathContainsIgnoredWord($fileName)
                || (!empty($this->parentFolderName) && mb_stripos($fileName, $this->parentFolderName) !== 0)
                || mb_strpos(basename($fileName), '.') === 0 // ignore hidden files
```

For consistency?

File: src/Backend/Modules/Pages/Ajax/UploadFile.php [hunk 3]
```diff
@@ -23,11 +23,17 @@ public function execute(): void
...
-        $fileName = $this->writeFile(
-            $this->getFileContentFromRequest($request),
-            $this->getFileNameFromRequest($request),
-            $request->get('type')
-        );
+        try {
+            $fileName = $this->writeFile(
+                $this->getFileContentFromRequest($request),
+                $this->getFileNameFromRequest($request),
+                $request->get('type')
```
Comment (jessedobbelaere) [inline src/Backend/Modules/Pages/Ajax/UploadFile.php:30]: These check warnings by codecov are quite annoying while code reviewing files without unit tests. I wonder if we can disable that somehow

File: src/Backend/Modules/Pages/Ajax/UploadFile.php [hunk 4]
```diff
@@ -100,6 +106,9 @@ private function writeFile(string $content, string $fileName, string $destinatio
         // convert the filename to url friendly version
         $baseName = Uri::getUrl(pathinfo($fileName, PATHINFO_FILENAME));
         $extension = pathinfo($fileName, PATHINFO_EXTENSION);
+        if (!in_array(strtolower($extension), ['jpg', 'jpeg', 'gif', 'png'])) {
```
Comment (jessedobbelaere) [inline src/Backend/Modules/Pages/Ajax/UploadFile.php:109]: That list is a bit restrictive.... Earlier this week I was using a user template block in Fork CMS and uploaded `.webp` and `.svg` files for example. `.avif` is a good one to have too in a year or so.

File: .htaccess (hunk 1 above)
Comment (carakas) [reply .htaccess:52]: I think it had to do with the index.php but I'll have a look if we can tighten it more

File: src/Backend/Modules/Pages/Ajax/UploadFile.php (hunk 4 above)
Comment (carakas) [reply src/Backend/Modules/Pages/Ajax/UploadFile.php:109]: I'll add those

File: src/Backend/Modules/Pages/Ajax/UploadFile.php (hunk 3 above)
Comment (carakas) [reply src/Backend/Modules/Pages/Ajax/UploadFile.php:30]: the patch ones are disabled so I don't know why they are showing

Comment (carakas) [reply src/Backend/Modules/Pages/Ajax/UploadFile.php:30]: 4bf38b485a718091613a7889ce78d3b650e19de9

File: src/Backend/Modules/Pages/Ajax/UploadFile.php (hunk 4 above)
Comment (carakas) [reply src/Backend/Modules/Pages/Ajax/UploadFile.php:109]: 0117c1e4e6c180ed134653066d1196745b31a43a

Comment (jessedobbelaere) [reply src/Backend/Modules/Pages/Ajax/UploadFile.php]: Awesome!

**Review by jessedobbelaere — APPROVED** [review]

**Pull request merged**